src/components/Tables/Table.tsx

```diff
@@ -0,0 +1,80 @@
+"use client";
+
+import React, { useState, useMemo, useEffect } from "react";
+import DataTable, { TableColumn } from "react-data-table-component";
+import TableFilterBar from "./TableFilterBar";
+
+interface Props {
+    data: { [key: string]: string | number | null }[];
+    headers: { [key: string]: string | null };
+}
+
+const Table: React.FC<Props> = (props) => {
+    const [filterText, setFilterText] = useState<{ [key: string]: string }>({});
+
+    // temporary solution to fix hydration error
+    const [domLoaded, setDomLoaded] = useState(false);
+    useEffect(() => {
+        setDomLoaded(true);
+    }, []);
+
+    const subHeaderComponentMemo = useMemo(() => {
```

Comment:
What is `useMemo` and why did you use this here, as opposed to having this be outside the component and passing in the state as props?

Reply:
I have moved it outside of the component.